cesium/delete.go

```diff
@@ -56,20 +57,31 @@ func (db *DB) DeleteChannel(ch ChannelKey) error {
 		return err
 	}
 
-	db.mu.Unlock()
+	// Rename the file first, so we can avoid hogging the mutex while deleting the directory
+	// may take a longer time.
+	// Rename the file to have a random suffix in case the channel is repeatedly created
+	// and deleted.
+	var newName = channelDirName(ch) + "-DELETE-" + strconv.Itoa(rand.Int())
+	err = db.fs.Rename(channelDirName(ch), newName)
```

Comment:
Minor note, but you call `channelDirName` twice. It has no impact on performance, but from a style perspective i'd make a new variable called `oldName` and call the function once.